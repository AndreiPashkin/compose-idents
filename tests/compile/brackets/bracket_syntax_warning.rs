//! Using deprecated bracket-based syntax should trigger a deprecation warning.
#![deny(warnings)]
use compose_idents::compose_idents;

compose_idents!(my_fn = [foo, _, "baz"], {
    fn my_fn() -> u32 {
        42
    }
},);

fn main() {
    assert_eq!(foo_baz(), 42);
}
