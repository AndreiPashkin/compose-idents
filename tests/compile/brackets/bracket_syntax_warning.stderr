error: use of deprecated function `foo_baz`: compose_idents!: Bracket-based syntax for alias definition, `alias = [arg1, func(arg2), ...]`, is deprecated, use expressions instead: `alias = concat(arg1, arg2, ...)`, `alias = arg`, `alias = func(arg)`, and so on
  --> tests/compile/brackets/bracket_syntax_warning.rs:12:16
   |
12 |     assert_eq!(foo_baz(), 42);
   |                ^^^^^^^
   |
note: the lint level is defined here
  --> tests/compile/brackets/bracket_syntax_warning.rs:2:9
   |
2  | #![deny(warnings)]
   |         ^^^^^^^^
   = note: `#[deny(deprecated)]` implied by `#[deny(warnings)]`
